backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -24,21 +24,30 @@ export default class TeamEventAttendanceDao {
     attendance: TeamEventAttendance
   ): Promise<TeamEventAttendance> {
     const teamAttendanceRef: DBTeamEventAttendance = {
-      member: memberCollection.doc(attendance.member.email),
-      ...(typeof attendance.hoursAttended === 'number'
-        ? { hoursAttended: attendance.hoursAttended }
-        : {}),
-      image: attendance.image,
-      eventUuid: attendance.eventUuid,
-      pending: attendance.pending,
-      uuid: attendance.uuid
+      ...attendance,
+      member: memberCollection.doc(attendance.member.email)
     };
-    await teamEventAttendanceCollection.doc(teamAttendanceRef.uuid).update(teamAttendanceRef);
+
+    const docRef = await teamEventAttendanceCollection.doc(teamAttendanceRef.uuid);
+    const currentDoc = await docRef.get();
+    const currentAttendance = currentDoc.data() as DBTeamEventAttendance;
+
+    if (
+      teamAttendanceRef.hoursAttended === undefined &&
+      currentAttendance.hoursAttended !== undefined
+    ) {
+      await docRef.update(
+        new admin.firestore.FieldPath('hoursAttended'),
+        admin.firestore.FieldValue.delete()
```

Comment:
@henry-li-06 @andrew032011 so here, after allowing undefined in firestore, if u originally `createteamevent attendance` and the `updateTeamEventAttendance` then the old `hoursAttended` value from the create remains, and doesnt  set to undefined. Instead of doing what i did here, is it ok, if I just set the value to zero if its undefined upon an update?

Reply:
+1 to Henry’s comment. I’d probably insert the specification (which I provided in the ticket) here too, but I’d modify it to say if the hoursAttended field is originally undefined, it cannot be overwritten afterwards.

Reply:
Can I include spec for the other 2

Reply:
> If I'm understanding your question correctly -- realistically speaking I don't think there would ever be a case where the TeamEventAttendance object has `hoursAttended` populated and then is later updated to be `undefined`.
> 
> 
> 
> I think in practice, the update dao method will mostly be used to approve TEC requests where only the `pending` field gets set to `true`.

Ahh I see